sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -448,7 +448,7 @@ private[execution] final class LongToUnsafeRowMap(val mm: TaskMemoryManager, cap
   private def nextSlot(pos: Int): Int = (pos + 2) & mask
 
   private def getRow(address: Long, resultRow: UnsafeRow): UnsafeRow = {
-    val offset = address >>> SIZE_BITS
+    val offset = (address >>> SIZE_BITS) + Platform.LONG_ARRAY_OFFSET
```

Comment:
This may sound a bit redundant. How creating two methods:
```scala
private[this] def toOffset(address: Long): Long = (address >>> SIZE_BITS) + Platform.LONG_ARRAY_OFFSET
private[this] def toAdress(offset: Long): Long = (offset - Platform.LONG_ARRAY_OFFSET) << SIZE_BITS
```
To make it clearer what you are doing there. They should be inlined, so the performance overhead is minimal.